zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/migration/MigrateMultiInstanceBodyTest.java

```diff
@@ -0,0 +1,704 @@
+
+    final var processInstanceKey = engine.processInstance().ofBpmnProcessId(processId).create();
+
+    assertThat(
+            RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_ACTIVATED)
+                .withElementId("A")
+                .limit(3))
+        .describedAs("Wait until all service tasks have activated")
+        .hasSize(3);
+
+    final var childProcessInstanceKey =
+        RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_ACTIVATED)
+            .withElementId("A")
+            .getFirst()
+            .getValue()
+            .getProcessInstanceKey();
+
+    // when
+    engine
+        .processInstance()
+        .withInstanceKey(processInstanceKey)
+        .migration()
+        .withTargetProcessDefinitionKey(targetProcessDefinitionKey)
+        .addMappingInstruction("callActivity1", "callActivity2")
+        .migrate();
+
+    // then
+    Assertions.assertThat(
+            RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_MIGRATED)
+                .withProcessInstanceKey(processInstanceKey)
+                .withElementType(BpmnElementType.MULTI_INSTANCE_BODY)
+                .getFirst()
+                .getValue())
+        .describedAs("Expect that process definition key is changed")
+        .hasProcessDefinitionKey(targetProcessDefinitionKey)
+        .describedAs("Expect that bpmn process id and element id changed")
+        .hasBpmnProcessId(targetProcessId)
+        .hasElementId("callActivity2")
+        .describedAs("Expect that version number did not change")
+        .hasVersion(1);
+
+    assertThat(
+            RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_MIGRATED)
+                .withProcessInstanceKey(processInstanceKey)
+                .withElementType(BpmnElementType.CALL_ACTIVITY)
+                .limit(3))
+        .extracting(Record::getValue)
+        .extracting(
+            r ->
+                tuple(
+                    r.getProcessDefinitionKey(),
+                    r.getBpmnProcessId(),
+                    r.getElementId(),
+                    r.getVersion()))
+        .containsExactly(
+            tuple(targetProcessDefinitionKey, targetProcessId, "callActivity2", 1),
+            tuple(targetProcessDefinitionKey, targetProcessId, "callActivity2", 1),
+            tuple(targetProcessDefinitionKey, targetProcessId, "callActivity2", 1));
+
+    engine.job().ofInstance(childProcessInstanceKey).withType("A").complete();
+
+    assertThat(
+            RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_ACTIVATED)
+                .withProcessInstanceKey(childProcessInstanceKey)
+                .withElementType(BpmnElementType.END_EVENT)
+                .withElementId("multi_instance_child_process_end")
+                .findAny())
+        .describedAs("Expect that the process instance is continued in the target process")
+        .isPresent();
```

Comment:
❌ This verifies that the end event of the child process instance has activated. However, the child process instance was not migrated. We need to verify that the parent process instance continues in the target process, e.g.:
- after all jobs are completed, the multi-instance body is completed and flows to a different element (one that didn't exist in the source).